Bottom: af4afd3216b9addcb5809a68e568b3213b6c03a1
Top:    89da7110d11eacd6bc5744af902e2e24e8a49539
Author: Ian Kent <raven@themaw.net>
Date:   2013-08-19 09:57:34 +0800

debug.patch


---

diff --git a/lib/parse_subs.c b/lib/parse_subs.c
index 2326838..cc1b4a4 100644
--- a/lib/parse_subs.c
+++ b/lib/parse_subs.c
@@ -35,6 +35,7 @@ static struct types map_type[] = {
 	{ "ldaps", 5 },
 	{ "hesiod", 6 },
 	{ "userdir", 7 },
+	{ "hosts", 5 },
 };
 static unsigned int map_type_count = sizeof(map_type)/sizeof(struct types);
 
diff --git a/modules/mount_autofs.c b/modules/mount_autofs.c
index 8c1e600..3d96e69 100644
--- a/modules/mount_autofs.c
+++ b/modules/mount_autofs.c
@@ -184,6 +184,7 @@ int mount_mount(struct autofs_point *ap, const char *root, const char *name,
 		return 1;
 	}
 	argv[0] = info->map;
+	error(LOGOPT_ANY, "info type %s format %s map %s", info->type, info->format, info->map);
 
 	if (options) {
 		p = options;
diff --git a/modules/parse_sun.c b/modules/parse_sun.c
index c1fc528..adf302a 100644
--- a/modules/parse_sun.c
+++ b/modules/parse_sun.c
@@ -881,7 +881,8 @@ static int validate_location(unsigned int logopt, char *loc)
 		    !strncmp(ptr, "file:", 5) || !strncmp(ptr, "yp:", 3) ||
 		    !strncmp(ptr, "nis:", 4) || !strncmp(ptr, "nisplus:", 8) ||
 		    !strncmp(ptr, "ldap:", 5) || !strncmp(ptr, "ldaps:", 6) ||
-		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4))
+		    !strncmp(ptr, "sss:", 4) || !strncmp(ptr, "dir:", 4) ||
+		    !strncmp(ptr, "hosts:", 4))
 			return 1;
 		error(logopt,
 		      "expected colon delimeter not found in location %s",
@@ -1294,6 +1295,16 @@ int parse_mount(struct autofs_point *ap, const char *name,
 
 	p = skipspace(pmapent);
 
+	/*
+	 * Change special map -hosts into <type>: format so it
+	 * doesn't confuse the options parse.
+	 */
+	if (strstr(p, "-hosts")) {
+		char *pos = p;
+		*pos++ = 'h'; *pos++ = 'o'; *pos++ = 's'; *pos++ = 't';
+		*pos++ = ':';
+	}
+
 	/* Deal with 0 or more options */
 	if (*p == '-') {
 		char *tmp, *mnt_options = NULL;
